app/src/marketplace/modules/contractProduct/services.js

```diff
@@ -304,7 +304,8 @@ export const getWhitelistAddresses = async (id: ProductId, usePublicNode: boolea
     return whitelist
 }
 
-export const isAddressWhitelisted = async (id: ProductId, address: Address) => {
-    const whitelist = await getWhitelistAddresses(id, true)
-    return whitelist.map((item) => item.status !== 'removed' && item.address.toLowerCase()).includes(address.toLowerCase())
+export const isAddressWhitelisted = async (id: ProductId, address: Address, usePublicNode: boolean = true) => {
+    const isWhitelistStatus = await call(contractMethods(usePublicNode).getWhitelistState(getValidId(id), address))
```

Comment:
@tumppi Changed this to use the actual smart contract call as it's a better source of truth than an event list (why not just include a getter for the list in the contract..? ). Let me know if this misses something